src/main/java/com/pubnub/api/managers/StateManager.java

```diff
@@ -79,12 +80,12 @@ public synchronized boolean handleOperation(final PubSubOperation... pubSubOpera
             if (pubSubOperation instanceof SubscribeOperation) {
                 if (adaptSubscribeBuilder((SubscribeOperation) pubSubOperation)) {
                     stateChanged = true;
-                    shouldAnnounce = true;
+                    announceStatus = PNStatusCategory.PNConnectedCategory;
                 }
             } else if (pubSubOperation instanceof UnsubscribeOperation) {
                 unsubscribe((UnsubscribeOperation) pubSubOperation);
                 stateChanged = true;
-                shouldAnnounce = true;
+                announceStatus = PNStatusCategory.PNConnectedCategory;
```

Comment:
We are notifying users that we're still connected after they unsubscribe from some channels assuming that at least one left. In case they unsubscribed from everything that won't have any effect

Reply:
Shouldn't it be `PNReconnectedCategory `instead of `PNConnectedCategory `?

Reply:
It was previously `PNConnectedCategory` so I kept it the same way